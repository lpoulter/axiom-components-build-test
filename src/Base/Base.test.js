import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import Base from './Base';

const getComponent = (props = {}) =>
  renderer.create(
    <Base { ...props } />
  );

describe('Base', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders string Component', () => {
    const component = getComponent({ Component: 'span' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders class Component', () => {
    class CustomComponent extends Component {
      render() {
        return (
          <span />
        );
      }
    }

    const component = getComponent({ Component: CustomComponent });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders functional Component', () => {
    const CustomComponent = () => <span />;
    const component = getComponent({ Component: CustomComponent });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders className', () => {
    const component = getComponent({ className: 'Lorem Ipsum' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders cloak', () =>
    [undefined, true, false].forEach((cloak) =>
      it(String(cloak), () => {
        const component = getComponent({ cloak });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      })
    )
  );

  it('renders cloakContainer', () => {
    const component = getComponent({ cloakContainer: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with hiddenUntil', () => {
    ['small', 'medium', 'large'].forEach((hiddenUntil) => {
      it(hiddenUntil, () => {
        const component = getComponent({ hiddenUntil });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with visibleUntil', () => {
    ['small', 'medium', 'large'].forEach((visibleUntil) => {
      it(visibleUntil, () => {
        const component = getComponent({ visibleUntil });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  it('renders sticky without style', () => {
    const component = getComponent({ sticky: '1rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders sticky with style', () => {
    const component = getComponent({ sticky: '1rem', style: { height: '1rem' } });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders container', () => {
    const component = getComponent({ container: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with textBreak', () => {
    ['none', 'all', 'word'].forEach((textBreak) => {
      it(textBreak, () => {
        const component = getComponent({ textBreak });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with textCenter', () => {
    [true, 'small', 'medium', 'large'].forEach((textCenter) => {
      it(String(textCenter), () => {
        const component = getComponent({ textCenter });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with textCase', () => {
    ['upper', 'capital', 'lower'].forEach((textCase) => {
      it(textCase, () => {
        const component = getComponent({ textCase });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });


  describe('renders with textColor', () => {
    [
      'day',
      'disabled',
      'error',
      'night',
      'subtle',
      'success',
      'warning',
    ].forEach((textColor) => {
      it(textColor, () => {
        const component = getComponent({ textColor });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  it('renders textEllipsis', () => {
    const component = getComponent({ textEllipsis: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textEmphasize', () => {
    const component = getComponent({ textEmphasize: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with textLeft', () => {
    [true, 'small', 'medium', 'large'].forEach((textLeft) => {
      it(String(textLeft), () => {
        const component = getComponent({ textLeft });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with textRight', () => {
    [true, 'small', 'medium', 'large'].forEach((textRight) => {
      it(String(textRight), () => {
        const component = getComponent({ textRight });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with textSize', () => {
    [
      'display2',
      'display1',
      'headline',
      'headtitle',
      'large',
      'label',
      'body',
      'small',
    ].forEach((textSize) => {
      it(textSize, () => {
        const component = getComponent({ textSize });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  it('renders textStrike', () => {
    const component = getComponent({ textStrike: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textStrong', () => {
    const component = getComponent({ textStrong: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textUnderline', () => {
    const component = getComponent({ textUnderline: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
