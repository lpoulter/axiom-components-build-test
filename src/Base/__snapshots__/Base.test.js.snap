// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders class Component 1`] = `<span />`;

exports[`Base renders className 1`] = `
<div
  className="Lorem Ipsum"
/>
`;

exports[`Base renders cloak false 1`] = `
<div
  className="ax-cloak ax-cloak--visible"
/>
`;

exports[`Base renders cloak true 1`] = `
<div
  className="ax-cloak"
/>
`;

exports[`Base renders cloak undefined 1`] = `
<div
  className=""
/>
`;

exports[`Base renders cloakContainer 1`] = `
<div
  className="ax-cloak__container"
/>
`;

exports[`Base renders container 1`] = `
<div
  className="ax-container"
/>
`;

exports[`Base renders functional Component 1`] = `<span />`;

exports[`Base renders sticky with style 1`] = `
<div
  className="ax-sticky"
  style={
    Object {
      "height": "1rem",
      "top": "1rem",
    }
  }
/>
`;

exports[`Base renders sticky without style 1`] = `
<div
  className="ax-sticky"
  style={
    Object {
      "top": "1rem",
    }
  }
/>
`;

exports[`Base renders string Component 1`] = `
<span
  className=""
/>
`;

exports[`Base renders textEllipsis 1`] = `
<div
  className="ax-text--ellipsis"
/>
`;

exports[`Base renders textEmphasize 1`] = `
<div
  className="ax-text--emphasize"
/>
`;

exports[`Base renders textStrike 1`] = `
<div
  className="ax-text--strike"
/>
`;

exports[`Base renders textStrong 1`] = `
<div
  className="ax-text--strong"
/>
`;

exports[`Base renders textUnderline 1`] = `
<div
  className="ax-text--underline-body"
/>
`;

exports[`Base renders with defaultProps 1`] = `
<div
  className=""
/>
`;

exports[`Base renders with hiddenUntil large 1`] = `
<div
  className="ax-hidden-until--large"
/>
`;

exports[`Base renders with hiddenUntil medium 1`] = `
<div
  className="ax-hidden-until--medium"
/>
`;

exports[`Base renders with hiddenUntil small 1`] = `
<div
  className="ax-hidden-until--small"
/>
`;

exports[`Base renders with textBreak all 1`] = `
<div
  className="ax-text--break-all"
/>
`;

exports[`Base renders with textBreak none 1`] = `
<div
  className="ax-text--break-none"
/>
`;

exports[`Base renders with textBreak word 1`] = `
<div
  className="ax-text--break-word"
/>
`;

exports[`Base renders with textCase capital 1`] = `
<div
  className="ax-text--case-capital"
/>
`;

exports[`Base renders with textCase lower 1`] = `
<div
  className="ax-text--case-lower"
/>
`;

exports[`Base renders with textCase upper 1`] = `
<div
  className="ax-text--case-upper"
/>
`;

exports[`Base renders with textCenter large 1`] = `
<div
  className="ax-text--align-center--large"
/>
`;

exports[`Base renders with textCenter medium 1`] = `
<div
  className="ax-text--align-center--medium"
/>
`;

exports[`Base renders with textCenter small 1`] = `
<div
  className="ax-text--align-center--small"
/>
`;

exports[`Base renders with textCenter true 1`] = `
<div
  className="ax-text--align-center"
/>
`;

exports[`Base renders with textColor day 1`] = `
<div
  className="ax-text--color-day"
/>
`;

exports[`Base renders with textColor disabled 1`] = `
<div
  className="ax-text--color-disabled"
/>
`;

exports[`Base renders with textColor error 1`] = `
<div
  className="ax-text--color-error"
/>
`;

exports[`Base renders with textColor night 1`] = `
<div
  className="ax-text--color-night"
/>
`;

exports[`Base renders with textColor subtle 1`] = `
<div
  className="ax-text--color-subtle"
/>
`;

exports[`Base renders with textColor success 1`] = `
<div
  className="ax-text--color-success"
/>
`;

exports[`Base renders with textColor warning 1`] = `
<div
  className="ax-text--color-warning"
/>
`;

exports[`Base renders with textLeft large 1`] = `
<div
  className="ax-text--align-left--large"
/>
`;

exports[`Base renders with textLeft medium 1`] = `
<div
  className="ax-text--align-left--medium"
/>
`;

exports[`Base renders with textLeft small 1`] = `
<div
  className="ax-text--align-left--small"
/>
`;

exports[`Base renders with textLeft true 1`] = `
<div
  className="ax-text--align-left"
/>
`;

exports[`Base renders with textRight large 1`] = `
<div
  className="ax-text--align-right--large"
/>
`;

exports[`Base renders with textRight medium 1`] = `
<div
  className="ax-text--align-right--medium"
/>
`;

exports[`Base renders with textRight small 1`] = `
<div
  className="ax-text--align-right--small"
/>
`;

exports[`Base renders with textRight true 1`] = `
<div
  className="ax-text--align-right"
/>
`;

exports[`Base renders with textSize body 1`] = `
<div
  className="ax-text--size-body"
/>
`;

exports[`Base renders with textSize display1 1`] = `
<div
  className="ax-text--size-display1"
/>
`;

exports[`Base renders with textSize display2 1`] = `
<div
  className="ax-text--size-display2"
/>
`;

exports[`Base renders with textSize headline 1`] = `
<div
  className="ax-text--size-headline"
/>
`;

exports[`Base renders with textSize headtitle 1`] = `
<div
  className="ax-text--size-headtitle"
/>
`;

exports[`Base renders with textSize label 1`] = `
<div
  className="ax-text--size-label"
/>
`;

exports[`Base renders with textSize large 1`] = `
<div
  className="ax-text--size-large"
/>
`;

exports[`Base renders with textSize small 1`] = `
<div
  className="ax-text--size-small"
/>
`;

exports[`Base renders with visibleUntil large 1`] = `
<div
  className="ax-visible-until--large"
/>
`;

exports[`Base renders with visibleUntil medium 1`] = `
<div
  className="ax-visible-until--medium"
/>
`;

exports[`Base renders with visibleUntil small 1`] = `
<div
  className="ax-visible-until--small"
/>
`;
